client/text.cpp

```diff
@@ -1459,23 +1474,63 @@ QString text_happiness_nationality(const struct city *pcity)
  */
 QString text_happiness_wonders(const struct city *pcity)
 {
-  struct effect_list *plist = effect_list_new();
-  QString str;
+  int effects_count = effect_list_size(get_effects(EFT_MAKE_HAPPY))
+                      + effect_list_size(get_effects(EFT_NO_UNHAPPY))
+                      + effect_list_size(get_effects(EFT_FORCE_CONTENT));
+  if (effects_count == 0) {
+    // Disabled in the ruleset
+    return QString();
+  }
 
-  get_city_bonus_effects(plist, pcity, nullptr, EFT_MAKE_HAPPY);
-  get_city_bonus_effects(plist, pcity, nullptr, EFT_FORCE_CONTENT);
-  get_city_bonus_effects(plist, pcity, nullptr, EFT_NO_UNHAPPY);
-  if (0 < effect_list_size(plist)) {
-    QString effects;
+  auto str = QStringLiteral("<p>");
+  bool wrote_something = false;
 
-    effects = get_effect_list_req_text(plist);
-    str = QString(_("Wonders: %1.")).arg(effects);
-  } else {
-    str = _("Wonders: None.");
+  int happy_bonus = get_city_bonus(pcity, EFT_MAKE_HAPPY);
+  if (happy_bonus > 0) {
+    if (happy_bonus == 1) {
+      str += _("Up to one content citizen in this city is made happy.");
+    } else {
+      str +=
+          QString(PL_("Up to %1 unhappy or content citizen in this city is "
+                      "made happy (unhappy citizens count double).",
+                      "Up to %1 unhappy or content citizens in this city "
+                      "are made happy (unhappy citizens count double).",
+                      happy_bonus))
+              .arg(happy_bonus);
+    }
+
+    // Add a list of active effects
+    str += QStringLiteral("</p><p>");
+    str += text_happiness_effect_details(pcity, EFT_MAKE_HAPPY);
+    str += QStringLiteral("</p><p>");
+    wrote_something = true;
   }
-  effect_list_destroy(plist);
 
-  return str.trimmed();
+  if (get_city_bonus(pcity, EFT_NO_UNHAPPY) > 0) {
+    str += _("No citizens in this city can ever be unhappy or angry. "
+             "Any unhappy or angry citizen is automatically made content.");
```

Comment:
Consider "... or angry citizen*s are* automatically ..." instead, to be consistent with the first sentence.